container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/backend_creator/backend_creator.go

```diff
@@ -86,7 +86,10 @@ func getRemoteDockerKurtosisBackend(
 	optionalApiContainerModeArgs *APIContainerModeArgs,
 	remoteBackendConfig *configs.KurtosisRemoteBackendConfig,
 ) (backend_interface.KurtosisBackend, error) {
-	remoteDockerClientOpts, err := buildRemoteDockerClientOpts(remoteBackendConfig)
+	remoteDockerClientOpts, cleanCertFilesFunc, err := buildRemoteDockerClientOpts(remoteBackendConfig)
+	if cleanCertFilesFunc != nil {
+		defer cleanCertFilesFunc()
+	}
```

Comment:
Here you can just do `defer cleanCertFilesFunc()` after the `if err != nil`.
If `err != nil`, we know the function won't be nil.